qcodes/config/qcodesrc_schema.json

```diff
@@ -82,10 +82,33 @@
                     "type": "string",
                     "default": "hot"
                 },
-                "smart_colorscale":{
-                    "description": "Use a smart colorscale, that scales such that outliers of the data will not be included in the min/max range.",
-                    "type": "boolean",
-                    "default": false
+                "auto_color_scale":{
+                    "type" : "object",
+                    "description": "Control of a auto color scale, that scales such that potential outliers of the data will not be included in the min/max range.",
+                    "properties" : {
+                        "enabled":{
+                            "description": "Enable automatic color scaling",
+                            "type": "boolean",
+                            "default": false
+                        },
+                        "cutoff_percentile":{
+                            "description": "Upper and lower percentage of datapoints that may maximally be discarded by the auto color scale. For example for [1,1] the auto color scale will in a set of 10 0000 points not clip more than the 100 points of lowest and highest value.",
+                            "type": "array",
+                            "items": {
+                                "type": "number",
+                                "type": "number"
+                            },
+                            "default": [0.5, 0.5]
+                        },
+                        "color_over":{
+                            "description": "Matplotlib color representing the datapoints clipped by the upper limit",
+                            "default": "white"
```

Comment:
why are the defaults here different from the defaults in the kwargs of the functions? what does it actually supposed to signify? (for example, that for qcodes users we recommend white/gray, and for users who implement custom plotting solutions that use our autoscaling functions should default to magenta/cyan?)

Reply:
So the colors here I chose different on purpose because we also define a default color scale in the config file and the over and under colors have to be matching the color scale. But thank you for asking, as I just realized that the default color map is hot and viridis. So white and black are extremely poor choices here.
Cyan an mangenta are kind of safe colors to use for any color scale ;-)

Reply:
I just realized that the colormap from the config is not being used in `plot_by_id` and added this here.